### ISSUE TITLE
"cut_down_prompt" causes infinite loop when in raw mode

### ISSUE THREAD

**Issue by marktaiwan** [thadunge2/AIDungeon#63]
When playing in raw mode, after the story get to a certain length, it would get stuck in a infinite loop:
https://github.com/thadunge2/AIDungeon/blob/c198371c34d914e9d996559ef850c87a76f572c4/generator/gpt2/gpt2_generator.py#L89-L90

This is because [`cut_down_prompt`](https://github.com/thadunge2/AIDungeon/blob/c198371c34d914e9d996559ef850c87a76f572c4/generator/gpt2/gpt2_generator.py#L130-L133) uses the player action indicator `>` to split and reduce the prompt, which doesn't exist in raw mode, leaving it unchanged.

There's probably a more elegant solution for this, but this is the workaround I'm using for my offline copy for the time being. It just reuses the code from #27 to cut down the prompt sentence by sentence if the generator is in raw mode.
```python
def cut_down_prompt(self, prompt):
    if not self.raw:
        split_prompt = prompt.split(">")
        expendable_text = ">".join(split_prompt[2:])
        return split_prompt[0] + (">" + expendable_text if len(expendable_text) > 0 else "")
    else:
        sentences = string_to_sentence_list(prompt.lstrip())
        sentences = sentences[1:]
        new_text = ""
        for i in range(len(sentences)):
            if sentences[i] == "<break>":
                new_text = new_text + "\n"
            else:
                new_text = new_text + " " + sentences[i]
        return new_text.lstrip()
```

**Comment by ShnitzelKiller** [comment]
I created the PR with your code for now... I was actually confused by cut down prompt, so I guess I didn't handle it right with raw mode. Why does it remove the _second_ to last prompt?

**Comment by marktaiwan** [comment]
> Why does it remove the _second_ to last prompt?

Yeah I think it's rather odd too.

**Comment by thadunge2** [comment]
Fixed with #64
